### PR TITLE
draft documentation for MASST+

### PR DESCRIPTION
**Pull request by chaosarium**
Draft documentation on how to use the the combined masst+ binary.

### CODE REVIEW

File: instructions.md [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+
+## The search types
+
+**Exact search** mode matches query spectra against database spectra based on shared peaks only. The index subcommand ingests files into the disk-based database index. 
+
+**Analog search** mode (also known as error-tolerant) matches query spectra against database spectra based on sahred and shifted peaks. The index subcommand ingests files into the disk-based database index.
+
+## Creating Index
+
+To get started, create an spectrum index for the desired search type. The paramater available for each search type differ slightly, but both requires a directory for the index and some spectra file to add to the index.
```
Comment (mustafa-guler) [inline instructions.md]: paramater -> parameters

File: instructions.md [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+You can always consult `masst_plus --help` for detailed description of the available commands and flags. 
+
+## The search types
+
+**Exact search** mode matches query spectra against database spectra based on shared peaks only. The index subcommand ingests files into the disk-based database index. 
+
+**Analog search** mode (also known as error-tolerant) matches query spectra against database spectra based on sahred and shifted peaks. The index subcommand ingests files into the disk-based database index.
+
+## Creating Index
```
Comment (mustafa-guler) [inline instructions.md:13]: Index -> Indices

File: instructions.md [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+```sh
+./masst_plus exact index [OPTIONS] --dir <DIR> --spectra-files <PATHS>...
+```
+or
+```sh
+./masst_plus exact index [OPTIONS] -d <DIR> -s <PATHS>...
+```
+
+Where `[OPTIONS]` could include the following. Note that `--peak-tol` and `--pepmass-tol` will be fixed for index. Flags under `Spectra` will be applied to preprocess the provided spectra files before they are added to the index, so they cannot be undone.
```
Comment (mustafa-guler) [inline instructions.md]: I would add that except for -d and -s users should feel free to leave all values at defaults if unsure

File: instructions.md [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+
+```sh
+./masst_plus analog index [OPTIONS] --dir <DIR> --spectra-files <PATHS>...
+```
+or
+```sh
+./masst_plus analog index [OPTIONS] -d <DIR> -s <PATHS>...
+```
+
+Where possible `[OPTIONS]` are as follows. Similar to exact index, `--peak-tol` will be fixed for index, though `--pepmass-tol` is not an option here. Flags under `Spectra` will be applied to preprocess the provided spectra files before they are added to the index, so they cannot be undone.
```
Comment (mustafa-guler) [inline instructions.md]: explain why pepmass-tol is excluded for ert mode

File: instructions.md [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+
+The tool comes with two types of search: exact and error-tolerant (analog). Separate indices are needed for each search type. This tool is thus split into two subcommands: exact and analog, each of which has its own set of subcommands for creating, adding to, and querying indices.
+
+You can always consult `masst_plus --help` for detailed description of the available commands and flags. 
+
+## The search types
+
+**Exact search** mode matches query spectra against database spectra based on shared peaks only. The index subcommand ingests files into the disk-based database index. 
+
+**Analog search** mode (also known as error-tolerant) matches query spectra against database spectra based on sahred and shifted peaks. The index subcommand ingests files into the disk-based database index.
```
Comment (mustafa-guler) [inline instructions.md]: combine with prev paragraph.

Have one sentecne for indexing, explaining a bit more that it pre-processes spectra into a format that's fast to query. For error tolerant explain taht the outputs will be pairs of spectra where one is shifted by some delta

File: instructions.md [hunk 6]
```diff
@@ -0,0 +1,352 @@
...
+```
+    
+</details>
+
+### Examples
+
+```sh
+./masst_plus exact index -d path/to/exact_index1 -s path/to/one_million_50.mgf
+```
+This creates an exact index for the spectra in `one_million_50.mgf` and adds it to the index at `path/to/exact_index1`, with the default options for spectrum preprocessing and filtering.
```
Comment (mustafa-guler) [inline instructions.md:228]: does this add to the index at -d or replace the index at -d?

File: instructions.md [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+
+```sh
+./masst_plus analog index -d path/to/analog_index1 -s path/to/one_million_49.mgf path/to/one_million_50.mgf
+``` 
+This creates an error tolerant index for the spectra in `one_million_49.mgf` and `one_million_50.mgf` and adds it to the index at `path/to/analog_index1`, with the default options for spectrum preprocessing and filtering.
```
Comment (mustafa-guler) [inline instructions.md]: same comment, does it add or replace?

File: instructions.md [hunk 8]
```diff
@@ -0,0 +1,352 @@
...
+``` 
+This adds the spectra in `one_million_0.mgf` and `one_million_1.mgf` to the analog index at `path/to/analog_index1`, with the default options for spectrum preprocessing and filtering.
+
+```sh
+./masst_plus exact add --no-filter --no-merge --normalize -i path/to/exact_index2 -s path/to/one_million_0.mgf
+``` 
+This adds the spectra in `one_million_0.mgf` to the exact index at `path/to/exact_index2`, with no peak filtering, no peak merging, and peak intensity normalization.
+
+## Query
+
```
Comment (mustafa-guler) [inline instructions.md:276]: missing section here?

File: instructions.md [hunk 9]
```diff
@@ -0,0 +1,352 @@
...
+### Result format
+
+The output will be in `csv` format. Each row of the `csv` output represent a match above the threshhold. The columns of the output represents:
+- `query_file` is the spectrum file where a query spectrum comes from
+- `query_scan` is a unique ID assigned to the query spectrum
+- `db_file` is the spectrum file where where the matched spectrum comes from
+- `db_scan` is a unique ID assigned to the matched spectrum
+- `score` is the similarity score between the query and matched spectrum
+
+An example output is as follows:
```
Comment (mustafa-guler) [inline instructions.md]: specificy the whitespace is added for readability just in the docs

**Review by mustafa-guler — CHANGES_REQUESTED** [review]

File: instructions.md [hunk 10]
```diff
@@ -0,0 +1,347 @@
...
+
+```sh
+./masst_plus ert score -i path/to/analog_index1 -s path/to/query1.mgf -t 0.7 -o path/to/output2.csv
+```
+This queries the analog index at `path/to/analog_index1` with the spectrum in `query1.mgf` and reports all matches above the threshold of 0.7 in `path/to/output.csv`.
+
+```sh
+./masst_plus exact score --no-filter --no-merge --normalize -i path/to/exact_index2 -s path/to/query1.mgf -t 0.8 -o path/to/output3.csv
+```
+This queries the exact index at `path/to/exact_index2` with the spectrum in `query1.mgf` and reports all matches above the threshold of 0.8 in `path/to/output.csv`, with no peak filtering, no peak merging, and peak intensity normalization.
```
Comment (chaosarium) [inline instructions.md]: output file name mismatch

File: instructions.md (hunk 6 above)
Comment (chaosarium) [reply instructions.md:228]: It will result in an error if there is already a index at -d

Comment (chaosarium) [reply instructions.md:228]: I'll add this to the docs

**Comment by chaosarium** [comment]
Whoops some of the later commits got automatically included in this pull request. They should be for release CI

**Pull request merged**